astropy/coordinates/sky_coordinate.py

```diff
@@ -1163,6 +1163,44 @@ def spherical_offsets_to(self, tocoord):
         dlat = acoord.spherical.lat.view(Angle)
         return dlon, dlat
 
+    def spherical_offsets_by(self, d_lon, d_lat):
```

Comment:
Would it be worth expanding this to also work for lon / colat with a flag, e.g. "iscolat=False"?
Alternatively, if ``d_lat`` were optional, then ``d_lon`` could be also be passed as a Representation object of any variety and just converted.

Reply:
Hm, I think the specialized need for dealing with spherical offsets is fairly astro-specific where the standard is lon/lat coordinate systems. For any other offset needs, I think users should really be using representations themselves...
Besides, this method is meant to be symmetric with `spherical_offsets_to()` which is specific to lon/lat offsets.